dnsmasq-metrics/pkg/server/options.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import "github.com/spf13/pflag"
+
+// Options for the daemon
+type Options struct {
+	DnsMasqPort           int
+	DnsMasqAddr           string
+	DnsMasqPollIntervalMs int
+
+	PrometheusAddr      string
+	PrometheusPort      int
+	PrometheusPath      string
+	PrometheusNamespace string
+	PrometheusSubsystem string
+}
+
+// NewOptions creates a new options struct.
+func NewOptions() *Options {
+	// Defaults should be set by ConfigureFlags() below.
+	return &Options{}
+}
+
+// ConfigureFlags adds command line options for the daemon
+func (opt *Options) ConfigureFlags(flagSet *pflag.FlagSet) {
```

Comment:
It's minor in this case, so maybe not worth doing, but a bit of thought on the pattern of flags in libs.  This couples the lib to a flag impl and menas that any app that uses this lib uses these flag names or else defines them all.  In this case we don't have another user, so not a huge deal.

A better pattern is to make a Config or Options struct in this package, with a NewConfig() func that fills in default values, and make the `main` package define the flags that populate the config.  Then pass the config down to Run().  You already do most of this.